runelite-client/src/main/java/net/runelite/client/plugins/account/AccountPlugin.java

```diff
@@ -57,24 +59,52 @@
 	@Inject
 	private ScheduledExecutorService executor;
 
-	private NavigationButton loginButton;
-	private NavigationButton logoutButton;
+	private JButton loginButton;
+	private JButton logoutButton;
 
 	@Override
 	protected void startUp() throws Exception
 	{
-		loginButton = new NavigationButton("Login", ImageIO.read(getClass().getResourceAsStream("login_icon.png")));
-		logoutButton = new NavigationButton("Logout", ImageIO.read(getClass().getResourceAsStream("logout_icon.png")));
+		loginButton = new JButton();
+		loginButton.setToolTipText("Login");
 		loginButton.addActionListener(this::loginClick);
+
+		logoutButton = new JButton();
+		logoutButton.setToolTipText("Logout");
 		logoutButton.addActionListener(this::logoutClick);
-		ui.getPluginToolbar().addNavigation(sessionManager.getAccountSession() != null ? logoutButton : loginButton);
+
+		addAndRemoveButtons();
+	}
+
+	private void addAndRemoveButtons()
+	{
+		try
+		{
+			TitleToolbar tb = ui.getTitleToolbar();
+			tb.remove(loginButton);
+			tb.remove(logoutButton);
+			if (sessionManager.getAccountSession() == null)
+			{
+				BufferedImage image = ImageIO.read(getClass().getResourceAsStream("login_icon.png"));
```

Comment:
These images can be loaded on startup instead of loading them every time session and plugin state changes.